numl/NeutrinoML/HDF5Maker_module.cc

```diff
@@ -273,7 +307,8 @@ void HDF5Maker::analyze(art::Event const& e)
   // Add invisible particles to hierarchy
   for (int id : g4id) {
     const MCParticle* p = pi->TrackIdToParticle_P(abs(id));
-    while (p->Mother() != 0) {
+    if (p==NULL) continue;
```

Comment:
under what circumstances is the particle pointer null? i worry this breaks a necessary condition for forming more sophisticated ground truths – an unbroken chain from each visible particle back to the generator primary. does uboone throw away intermediate particles in its particle inventory scheme?

Reply:
this was a somewhat rare occurrence (I think ~50% of the jobs were failing, but you need to account for tens of events per jobs and all MCPartices in the event). I vaguely recall that some intermediate particles are dropped but I do not recall the details. I would not expect this to be breaking the ground truth in a serious way, but we should check